app/src/main/kotlin/com/wire/android/ui/common/SearchBar.kt

```diff
@@ -0,0 +1,167 @@
+import androidx.compose.animation.ExperimentalAnimationApi
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.LocalContentAlpha
+import androidx.compose.material.LocalContentColor
+import androidx.compose.material.OutlinedTextField
+import androidx.compose.material.Text
+import androidx.compose.material.TextFieldDefaults
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.material.icons.filled.Search
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment.Companion.CenterHorizontally
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.focus.onFocusChanged
+import androidx.compose.ui.graphics.Color.Companion.Transparent
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.wire.android.R
+import com.wire.android.ui.theme.WireColor
+import com.wire.android.ui.theme.WireLightColors
+
+@ExperimentalAnimationApi
+@ExperimentalComposeUiApi
+@Composable
+fun SearchBarFullScreen(
+    searchText: String,
+    placeholderText: String = "",
+    onSearchTextChanged: (String) -> Unit = {},
+    onClearClick: () -> Unit = {},
+    onNavigateBack: () -> Unit = {},
+    matchesFound: Boolean,
+    modifier: Modifier = Modifier,
+    results: @Composable () -> Unit = {}
+) {
+    var showClearButton by remember { mutableStateOf(false) }
+    val keyboardController = LocalSoftwareKeyboardController.current
+    val focusRequester = remember { FocusRequester() }
+
+    TopAppBar(title = { Text("") },
+        backgroundColor = WireLightColors.background,
```

Comment:
I think we should not refer directly to `WireLightColors`, instead use `MaterialTheme.colors.background` - we set `WireTheme` for the whole `WireApp` and this theme decides itself if it should use light or dark colors, using `MaterialTheme.colors.xxx` we get the proper one and don't have to decide again if it should be light or dark
Maybe `WireLightColors` and `WireDarkColors` should be private
I can clean this up after I complete the login screen UI, I actually thought about it, let me know if you agree 😄

Reply:
I agree on that :P we should try to use the MaterialTheme.color thingy where we can and if not maybe there is some way to support choosing the color depending on dark/light theme

Reply:
Yepp, it makes total sense. I already moved the light and dark colors palette to the Theme file and declare them as private. If you think of a better way of organising it, let me know or add it in your future PR.